subs/pantry/src/Pantry/Repo.hs

```diff
@@ -71,25 +71,25 @@ getRepo' repo@(Repo url commit repoType' subdir) pm =
         dir = tmpdir </> suffix
         tarball = tmpdir </> "foo.tar"
 
-    let (commandName, cloneArgs, resetArgs, archiveArgs) =
+    let (commandName, resetArgs, submoduleArgs, archiveArgs) =
           case repoType' of
             RepoGit ->
               ( "git"
-              , ["--recursive"]
               , ["reset", "--hard", T.unpack commit]
+              , ["submodule", "update", "--init", "--recursive"]
               , ["archive", "-o", tarball, "HEAD"]
               )
             RepoHg ->
               ( "hg"
-              , []
               , ["update", "-C", T.unpack commit]
+              , []
```

Comment:
Doesn't this mean that, for Mercurial, we'll be calling `hg` here with no arguments? I think using a `Nothing` value here and checking for that below would be appropriate.

Reply:
👍 Good suggestion. There's definitely no reason to have an extra shell out to `hg` with no arguments, I pushed another commit to fix that.

Also happy to squash things up back to a single commit, but I might wait for further feedback and do that once at the end.